src/web/app/pages-instructor/instructor-session-edit-page/copy-questions-from-other-sessions-modal/copy-questions-from-other-sessions-modal.component.ts

```diff
@@ -20,52 +24,139 @@ export class CopyQuestionsFromOtherSessionsModalComponent implements OnInit {
+        };
+        model.questionsTableRowModels.push(questionToCopy);
+      });
+      model.hasQuestionsLoaded = true;
+    }, (resp: ErrorMessageOutput) => {
+      model.hasLoadingFailed = true;
+      this.statusMessageService.showErrorToast(resp.error.message);
+    });
+  }
+
   /**
    * Gets the selected questions to copy.
    */
   getSelectedQuestions(): FeedbackQuestion[] {
-    return this.questionToCopyCandidates
-        .filter((c: QuestionToCopyCandidate) => c.isSelected)
-        .map((c: QuestionToCopyCandidate) => c.question);
+    const selectedQuestions: FeedbackQuestion[] = [];
+    this.feedbackSessionTabModels.forEach((model: FeedbackSessionTabModel) => {
+      if (model.questionsTableRowModels.length > 0) {
+        selectedQuestions.push(
+          ...model.questionsTableRowModels
+            .filter((c: QuestionToCopyCandidate) => c.isSelected)
+            .map((c: QuestionToCopyCandidate) => c.question),
+        );
+      }
+    });
+    return selectedQuestions;
+  }
+
+  /**
+   * Checks the option selected to sort feedback sessions.
+   */
+  isSelectedForSorting(by: SortBy): boolean {
+    return this.feedbackSessionTabModelsSortBy === by;
+  }
+
+  /**
+   * Sorts the list of feedback sessions.
+   */
+  sortFeedbackSessionTabs(by: SortBy): void {
+    this.feedbackSessionTabModelsSortBy = by;
+    this.feedbackSessionTabModels.sort(this.sortFeedbackSessionsBy(by));
   }
 
   /**
-   * Sorts the list of questions to copy.
+   * Sorts the list of questions for a feedback session.
    */
-  sortQuestionToCopyCandidates(by: SortBy): void {
-    this.candidatesSortBy = by;
+  sortQuestionsToCopyForFeedbackSession(model: FeedbackSessionTabModel, by: SortBy): void {
+    model.questionsTableRowModelsSortBy = by;
     // reverse the sort order
-    this.candidatesSortOrder =
-        this.candidatesSortOrder === SortOrder.DESC ? SortOrder.ASC : SortOrder.DESC;
-    this.questionToCopyCandidates.sort(this.sortCandidatesBy(by, this.candidatesSortOrder));
+    model.questionsTableRowModelsSortOrder =
+      model.questionsTableRowModelsSortOrder === SortOrder.DESC ? SortOrder.ASC : SortOrder.DESC;
+    model.questionsTableRowModels.sort(this.sortQuestionsBy(by, model.questionsTableRowModelsSortOrder));
   }
 
   /**
-   * Generates a sorting function.
+   * Generates a sorting function for sessions.
    */
-  protected sortCandidatesBy(by: SortBy, order: SortOrder):
-      ((a: QuestionToCopyCandidate, b: QuestionToCopyCandidate) => number) {
-    return ((a: QuestionToCopyCandidate, b: QuestionToCopyCandidate): number => {
+  protected sortFeedbackSessionsBy(by: SortBy):
+      ((a: FeedbackSessionTabModel, b: FeedbackSessionTabModel) => number) {
+    return ((a: FeedbackSessionTabModel, b: FeedbackSessionTabModel): number => {
       let strA: string;
       let strB: string;
+      let order: SortOrder;
       switch (by) {
+        case SortBy.COURSE_ID:
+          strA = String(a.courseId);
```

Comment:
Can you check if we are using `String()` or `.toString()` in the codebase? Just to make things consistent

Reply:
I've seen both instances used, but I think `.toString()` is more frequently used, so I have refactored the code to use `.toString()` instead!